pontoon/base/utils.py

```diff
@@ -642,29 +646,34 @@ def wrapper(request, *args, **kwargs):
             # If slug is still not defined, raise an error.
             if slug is None:
                 raise ValueError("Cannot extract slug from view arguments")
+            # Try to find a project using the current slug
+            try:
+                Project.objects.get(slug=slug)
+                # If found, continue with the original view function
+                return view_func(request, *args, **kwargs)
```

Comment:
This is better, but still adds a project lookup to each request that's then discarded. Is there a way to do as I suggested earlier, and to include this redirection in the error handling rather than proactively?

In each of the three functions that use this decorator, there's a call to `get_object_or_404()` for something like `Project.objects.get(slug=slug)`. Could that call be changed to trigger this flow, instead of the current decorator approach?